projects/demo-cypress/src/tests/swipe.cy.ts

```diff
@@ -0,0 +1,95 @@
+import {Component} from '@angular/core';
+import {TuiSwipe, TuiSwipeModule} from '@taiga-ui/cdk';
+import {TuiRootModule} from '@taiga-ui/core';
+
+describe('TuiSwipe', () => {
+    let component: TestComponent;
+
+    @Component({
+        template: `
+            <tui-root>
+                <section (tuiSwipe)="onSwipe($event)"></section>
+            </tui-root>
+        `,
+    })
+    class TestComponent {
+        swiped = '';
+
+        onSwipe(swipe: TuiSwipe): void {
+            this.swiped = swipe.direction;
+        }
+    }
+
+    beforeEach(() =>
+        cy
+            .mount(TestComponent, {
+                imports: [TuiRootModule, TuiSwipeModule],
+            })
+            .then(wrapper => {
+                component = wrapper.component;
+            }),
+    );
+
+    it('emits events bottom', () => {
+        cy.get('section')
+            .then(query => {
+                const element = query.get(0);
+
+                sendTouchEvent(0, 0, element, 'touchstart', 1);
+                sendTouchEvent(0, 100, element, 'touchend', 1);
+            })
+            .then(() => expect(component.swiped).to.eql('bottom'));
+    });
+
+    it('emits events right', () => {
+        cy.get('section')
+            .then(query => {
+                const element = query.get(0);
+
+                sendTouchEvent(0, 0, element, 'touchstart', 1);
+                sendTouchEvent(100, 0, element, 'touchend', 1);
+            })
+            .then(() => expect(component.swiped).to.eql('right'));
+    });
+
+    it('does not emit events due to threshold', () => {
+        cy.get('section')
+            .then(query => {
+                const element = query.get(0);
+
+                sendTouchEvent(0, 0, element, 'touchstart', 1);
+                sendTouchEvent(0, 20, element, 'touchend', 1);
+            })
+            .then(() => expect(component.swiped).to.eql(''));
+    });
+});
+
+function sendTouchEvent(
+    x: number,
+    y: number,
+    element: HTMLElement,
+    eventType: 'touchend' | 'touchstart',
+    identifier: number,
+): void {
+    const touchObj = new Touch({
+        identifier,
+        target: element,
+        clientX: x,
+        clientY: y,
+        radiusX: 2.5,
+        radiusY: 2.5,
+        rotationAngle: 10,
+        force: 0.5,
+    });
+
+    const touchEvent = new TouchEvent(eventType, {
+        cancelable: true,
+        bubbles: true,
+        touches: [touchObj],
+        targetTouches: [],
+        changedTouches: [touchObj],
+        shiftKey: true,
+    });
+
+    element.dispatchEvent(touchEvent);
```

Comment:
The same https://github.com/taiga-family/taiga-ui/pull/6642#discussion_r1476027194